lib/oscap_tests.pm

```diff
@@ -123,186 +711,541 @@ sub upload_logs_reports {
+        }
+    }
+    else {
+        record_info("No expected results", "No expected results found in \nfile: $expected_results_file_name\n for profile_ID: $profile_ID\ntype: $type\narch: $arch\n Using results defined in the test file.");
+        my @eval_match = ();
+        $_[0] = \@eval_match;
+    }
+    return $found;
+}
+
+sub get_test_exclusions {
+    # Get exclusions from remote file
+    my $exclusions = ();
+    my $found = -1;
+    my $type = "";
+    my $arch = "";
+
+    # If set in configuration to not use excusions
+    if ($use_exclusions == 0) {
+        return $found;
+    }
+    else {
+        if ($ansible_remediation == 1) {
+            $type = 'ansible';
+        }
+        else {
+            $type = 'bash';
+        }
+        if (is_s390x) { $arch = "s390x"; }
+        if (is_aarch64 or is_arm) { $arch = "aarch64"; }
+        if (is_ppc64le) { $arch = "ppc"; }
+        if (is_x86_64) { $arch = "x86_64"; }
+        # $sle_version and $profile_ID are global varables
+        my $exclusions_list_name = $sle_version . "-exclusions_list";
+        my $exclusions_file_name = "openqa_tests_exclusions.yaml";
+        my $url = "https://gitlab.suse.de/seccert-public/compliance-as-code-compiled/-/raw/main/content/";
+
+        download_file_from_https_repo($url, $exclusions_file_name);
+        upload_logs("$exclusions_file_name") if script_run "! [[ -e $exclusions_file_name ]]";
+        my $data = script_output("cat $exclusions_file_name", quiet => 1);
+
+        # Pharse the expected results
+        my $exclusions_data = YAML::PP::Load($data);
+        record_info("Looking exclusions", "Looking exclusions for \nprofile_ID: $profile_ID\ntype: $type\narch: $arch");
+
+        $exclusions = $exclusions_data->{$profile_ID}->{$type}->{$arch}->{$exclusions_list_name};
+        # If results defined
+        if (defined $exclusions) {
+            $found = 1;
+            if (not defined $$exclusions[0]) {
+                my @exclusions = ();
+                $_[0] = \@exclusions;
+                record_info("Got exclusions", "Got exclusions for \nprofile_ID: $profile_ID\ntype: $type\narch: $arch\nList excluded rules:\n @exclusions");
+            }
+            else {
+                $_[0] = $exclusions;
+                record_info("Got exclusions", "Got exclusions for \nprofile_ID: $profile_ID\ntype: $type\narch: $arch\nList of excluded rules:\n" . (join "\n", @$exclusions));
+            }
+        }
+        else {
+            record_info("No exclusions", "No exclusions found in \nfile: $exclusions_file_name\n for profile_ID: $profile_ID\ntype: $type\narch: $arch");
+            my @exclusions = ();
+            $_[0] = \@exclusions;
+        }
+        return $found;
+    }
+}
 
+sub oscap_security_guide_setup {
+    # Main test setup function
     select_console 'root-console';
+    # Setting $full_ansible_file_path aftr got ansible_profile_ID from test
```

Comment:
better not introduce names of variables in the comments, because variable names can changes in the future and the comments might be outdated easily.
Additionally I would like to mention that we don't have any code style for having a comment for each block or line, normally a comment is only needed if there is an exception or something that is not obvious by the code or require some experience in the area. For example this comment is obvious, it is not needed, similar above for `#Main setup function` which is attached to the console selection, that should be out, same than for a `zypper in` and you wil find more example when reading the code. It helps to readability as well to have comments only where they are needed.

Reply:
I am writing code and comments as much as possible commented and understandable.
Will cleanup possibly unneeded comments.